web_widget_x2many_graph/static/src/js/web_widget_x2many_graph.js

```diff
@@ -18,86 +18,43 @@
 //   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //
 //############################################################################
+odoo.define('web.web_widget_x2many_graph', function (require) {
+"use strict";
 
-openerp.web_widget_x2many_graph = function(instance)
-{
-    instance.web.form.widgets.add(
-        'x2many_graph',
-        'instance.web_widget_x2many_graph.FieldX2ManyGraph');
+    var core = require('web.core'),
+        form_common = require('web.form_common'),
+        QWeb = core.qweb;
 
-    instance.web_widget_x2many_graph.FieldX2ManyGraph = instance.web.form.AbstractField.extend( {
-        template: 'FieldX2ManyGraph',
-        widget_class: 'oe_form_field_x2many_graph',
-        field_x: 'sequence',
-        field_label_x: 'Sequence',
-        init: function (field_manager, node) {
-            this._super(field_manager, node);
-            this.field_x = node.attrs.field_x ? node.attrs.field_x : this.field_x;
-            this.field_label_x = node.attrs.field_label_x ? node.attrs.field_label_x : this.field_label_x;
-            this.dataset = new instance.web.form.One2ManyDataSet(this, this.field.relation);
-            this.dataset.o2m = this;
-            this.dataset.parent_view = this.view;
-            this.dataset.child_name = this.name;
-            this.set_value([]);
-            this.attrs = node.attrs;
-        },
-        start: function() {
-            this._super.apply(this, arguments);
-        },
-        get_value: function () {
-            var value = this.get('value');
-            return value
-        },
+    var FieldX2ManyGraphWidget = form_common.AbstractField.extend(form_common.ReinitializeFieldMixin, {
```

Comment:
Call it json-graph or something like that due to the refactor this is not anymore a X2MAnyGraph